fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java

```diff
@@ -1220,6 +1214,63 @@ private void updateDBRunningTxnNum(TransactionStatus preStatus, TransactionState
         return infos;
     }
     
+    // get show info of a specified txnId
+    public List<List<String>> getSingleTranInfo(long dbId, long txnId) throws AnalysisException {
+        List<List<String>> infos = new ArrayList<List<String>>();
+        readLock();
+        try {
+            Database db = Catalog.getInstance().getDb(dbId);
+            if (db == null) {
+                throw new AnalysisException("Database[" + dbId + "] does not exist");
+            }
+            
+            TransactionState txnState = idToTransactionState.get(txnId);
+            if (txnState == null) {
+                throw new AnalysisException("transaction with id " + txnId + " does not exist");
+            }
+            
+            if (ConnectContext.get() != null) {
```

Comment:
precondition?

Reply:
This method may be called internal, not in user connection thread. So I just add a judgement here.